crates/mzoon/src/wasm_opt.rs

```diff
@@ -97,23 +100,31 @@ async fn check_wasm_opt() {
 }
 
 #[throws]
-fn unpack_wasm_opt(tar_gz: Vec<u8>) {
+async fn unpack_wasm_opt(tar_gz: Vec<u8>) {
```

Comment:
The code is starting to smell a bit (too much `unwrap`s, hidden OS-dependent code, etc.) so I've tried to rewrite the function to the code below, what do you think?
(Tested only on Windows for now.)
```rust
#[throws]
async fn unpack_wasm_opt(tar_gz: Vec<u8>) {
    let tar = GzDecoder::new(tar_gz.as_slice());
    let mut archive = Archive::new(tar);

    for entry in archive.entries()? {
        let mut entry = entry?;
        let path = entry.path()?;

        let file_name = path
            .file_name()
            .ok_or_else(|| anyhow!("Entry without a file name"))?
            .to_str()
            .ok_or_else(|| anyhow!("Entry with a non-Unicode file name"))?;

        let output_dir = match file_name {
            // Windows | Linux + Mac
            "wasm-opt.exe" | "wasm-opt" => "frontend/binaryen/bin",
            // The lib is required on Mac.
            // Note: It's called `binaryen.lib` on Windows and `libbinaryen.a` on Linux.
            "libbinaryen.dylib" => "frontend/binaryen/lib",
            _ => continue,
        };
        create_dir_all(output_dir)?;
        entry.unpack(format!("{output_dir}/{file_name}"))?;
    }

    if let Err(error) = check_wasm_opt().await {
        eprintln!("wasm-opt installation failed: {error:#}");
    }
}
```
Also we can then remove the line `use std::path::Path;` from the top of the file.

Reply:
It looks better! Now only the match arm mentions the OS-dependency which is comfortable to me👍

Reply:
Ok! I've just tested the snippet above with Linux so it should work on both Linux and Windows.
Please update the code (the snippet above + remove that line with `Path`), test it with your Mac and then I'll merge the PR :)
Thanks!